### ISSUE TITLE
Cannot enable Mirrored network on WSL 

### ISSUE THREAD

**Issue by ecomter** [microsoft/WSL#11154]
### Windows Version

Microsoft Windows [Version 10.0.26052.1000]

### WSL Version

2.1.1.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.15.146.1-microsoft-standard-WSL2

### Distro Version

Ubuntu 18.04

### Repro Steps

I set the network type to mirrored in the .wslconfig file.

### Expected Behavior

Everything's getting good. It should work.

### Actual Behavior

When I start wsl it reports an error:

```
无法初始化Mirrored网络。若要切换回NAT网络请从 C:\Users\ecomter\.wslconfig中删除“experimental.networkingMode”
Error code: Wsl/Service/CreateInstance/CreateVm/ConfigureNetworking/0x803b0015
```

**Comment by chanpreetdhanjal** [comment]
Hi. Can you please collect networking logs by following the instructions below?
https://github.com/microsoft/WSL/blob/master/CONTRIBUTING.md#collect-wsl-logs-for-networking-issues

**Comment by ecomter** [comment]
I have finished it in the title “Diagnostic Logs” of the first answer.
[WslNetworkingLogs-2024-02-14_17-09-15.zip](https://github.com/microsoft/WSL/files/14277610/WslNetworkingLogs-2024-02-14_17-09-15.zip)

**Comment by ecomter** [comment]
Is there a available solution yet, please? I found in the Microsoft documentation that it might mean HCN_E_REQUEST_UNSUPPORTED error which is "The specified request is unsupported". I am hosting my virtual machine in azure, hope this helps!

**Comment by keith-horton** [comment]
Hi there. can you clarify what version of Windows is running in your virtual machine? Is this a server SKU? is it a headless server SKU? I'm reading code where it looks like it's blocking it for some reason.

**Comment by ecomter** [comment]
It's Windows Server 2025 Datacenter Azure Edition Build 26052 running on Microsoft Azure, I connect to it by RDP. So I think it is a Headless server SKU

**Comment by keith-horton** [comment]
Yeah, sorry. I found code in another component that's blocking Mirrored mode from running on Server SKUs :(

I'm going to try and fix this for the next release. I will have to pitch this to folks who work on servicing for existing releases.

**Comment by L2** [comment]
> Yeah, sorry. I found code in another component that's blocking Mirrored mode from running on Server SKUs :(
> 
> I'm going to try and fix this for the next release. I will have to pitch this to folks who work on servicing for existing releases.

Hi @keith-horton , just a friendly ping to see if you were able to get Mirrored mode working with Windows Server (2022 in particular) and if there's any eta on the release for the next WSL pre-release build. Thanks!

**Comment by joshmouch** [comment]
> It's Windows Server 2025 Datacenter Azure Edition Build 26052 running on Microsoft Azure, I connect to it by RDP. So I think it is a Headless server SKU

I'm also getting this on Windows Datacenter Server 2025 preview.

**Comment by captainhook** [comment]
I have settled with port forwarding for now as detailed in https://github.com/microsoft/WSL/issues/4150

Hopefully this constraint is removed for Windows Server SKUs soon.

**Comment by FrostleafChan** [comment]
Still getting this on Windows Server 2025 Version 24H2 Build 26100.2033.
` CreateInstance/CreateVm/ConfigureNetworking/0x803b0015 `, falling back to networkingMode=none.

**Comment by Vossiewos** [comment]
> Still getting this on Windows Server 2025 Version 24H2 Build 26100.2033. `CreateInstance/CreateVm/ConfigureNetworking/0x803b0015`, falling back to networkingMode=none.

Same. I'm hoping to run an up-to-date squid version under WSL, but in NAT mode, all requests logged are comming from the default-gateway of the WSL subnet instead of the true client, so there is no distinction between requests. (it's not only DNAT-ting on the Windows host, but also SNAT-ting) really was hoping that Mirrored Network mode would solve this for me.

**Comment by arixmkii** [comment]
Have same issue with new install of Server 2025 and WSL 2.4.10

**Comment by firezym** [comment]
Have same issue with new install of Server 2025 `26100.3194` and WSL2 `2.4.11.0`

**Comment by MJRT** [comment]
> Have same issue after updated to win11 24H2(26100.3194)

Changing the networking mode to VirtioProxy and restarting Windows works for me!

**Comment by fyq163** [comment]
mine even have more errors

> `wsl: 指定的请求不受支持。
Error code: CreateInstance/CreateVm/ConfigureNetworking/0x803b0015
wsl: Failed to configure network (networkingMode Mirrored), falling back to networkingMode None.
wsl: A localhost proxy configuration was detected but not mirrored into WSL. WSL in NAT mode does not support localhost proxies.`

**Comment by kidfruit** [comment]
Same here. Server 2025 datacenter 26100.3476 and wsl 2.4.12

**Comment by acdzh** [comment]
Have same issue with new install of Server 2025 `26110.4061` and WSL `2.4.13.0`

**Comment by windmill419** [comment]
Have same issue with new install of Server 2025 26110.4349 and WSL 2.5.7.0

**Comment by marclevine** [comment]
Have same issue with new install of Server 2025 26110.4652 and WSL 2.5.9.0 virtioproxy is working well

**Comment by yanshiqwq** [comment]
Have same issue with new install of Server 2025 `26100.4946` and WSL `2.5.10.0`

**Comment by Aethreverology** [comment]
Is there any update on this being fixed? It looks like it's been an issue for over a year. I'm still having it on Windows Server 2025.

**Comment by yanshiqwq** [comment]
> Have same issue with new install of Server 2025 `26100.4946` and WSL `2.5.10.0`

Have same issue with new install of Server 2025 `26100.6584` and WSL `2.6.1.0`

**Comment by mccaya** [comment]
Its not supported and depreciated.

**Comment by intelljames** [comment]
Have same issue with new install of Server 2025 26100.6584 and WSL 2.6.1.0

**Comment by yanshiqwq** [comment]
Have same issue with new install of Server 2025 26100.7462 and WSL 2.6.3.0

**Comment by elhananjair** [comment]
Any updates on this issue? I think starting from WSL2, many of us have started using WSL regularly on Windows Server, it would be best if this feature works.

**Comment by yanshiqwq** [comment]
Have same issue with new install of Server 2025 26100.32230 and WSL 2.7.0.0
😇

**Comment by hoshiizumiya** [comment]
shit work with the lastest windows 2025 and wsl2 version

**Comment by PPicku** [comment]
Same question on me.  I have Windows 2025 24H2 and WSL 2.7.3.0.

**Comment by LaoCaocao666** [comment]
Windows service 2025 24H2  WSL 2.7.3   Same question

**Comment by paasi6666** [comment]
Any updates to this?

**Comment by dem1995** [comment]
Same question

Edition	Windows 11 Pro
Version	23H2
Installed on	‎6/‎15/‎2026
OS build	22631.6199
Experience	Windows Feature Experience Pack 1000.22700.1108.0

WSL version: 2.7.8.0
Kernel version: 6.18.33.1-1
WSLg version: 1.0.73.2
MSRDC version: 1.2.6676
Direct3D version: 1.611.1-81528511
DXCore version: 10.0.26100.1-240331-1435.ge-release
Windows version: 10.0.22631.6199

**Comment by yanshiqwq** [comment]
Same issue，2026/07/12，
😇

WSL 版本: 2.7.10.0
内核版本: 6.18.33.2-2
WSLg 版本: 1.0.73.2
MSRDC 版本: 1.2.6676
Direct3D 版本: 1.611.1-81528511
DXCore 版本: 10.0.26100.1-240331-1435.ge-release
Windows: 10.0.26100.32690

<img width="796" height="347" alt="Image" src="https://github.com/user-attachments/assets/5361605b-9f3d-4b11-a144-2653b76a70d4" />

**Comment by yanshiqwq** [comment]
Could this be fixed before the release of Windows 12? 😂

**Comment by herepost** [comment]
> 是的，抱歉。我在另一个组件里发现了阻止镜像模式在服务器SKU上运行的代码:(
> 
> 我打算在下一个版本里尝试修复这个问题。我得向那些为现有发布做服务的人推介这个。

很抱歉，当时我希望了解这个问题的修复进度。感谢

**Comment by dorioku** [comment]
same 
WSL 版本: 2.7.11.0
内核版本: 6.18.33.2-2
WSLg 版本: 1.0.73.2
MSRDC 版本: 1.2.7214
Direct3D 版本: 1.611.1-81528511
DXCore 版本: 10.0.26100.1-240331-1435.ge-release
Windows: 10.0.26100.33158

wsl: 指定的请求不受支持。
错误代码: CreateInstance/CreateVm/ConfigureNetworking/0x803b0015
wsl: 无法配置网络 (networkingMode Mirrored)，回退到 networkingMode None。

**Comment by vibbix** [comment]
```
wsl --shutdown
Disable-NetAdapter -Name 'Ethernet 4' -Confirm:$false
Start-Sleep 3
wsl -d Ubuntu-24.04 -e true      # rebuilds FSE against loopback only
Enable-NetAdapter -Name 'Ethernet 4' -Confirm:$false
Start-Sleep 8
wsl --shutdown
```
I had to disable and re-enable my nic (no restart needed), and then it worked
